src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentgenerator/DocumentConversionServiceTest.java

```diff
@@ -91,4 +107,12 @@ public void getConvertedFilename() {
         assertThat(documentConversionService.getConvertedFilename("nodot"), is("nodot.pdf"));
         assertThat(documentConversionService.getConvertedFilename("word.docx"), is("word.pdf"));
     }
+
+    private byte[] loadResource(String testPdf) throws IOException {
+
+        try (InputStream resourceAsStream = getClass().getResourceAsStream(testPdf)) {
+            assert resourceAsStream != null;
+            return resourceAsStream.readAllBytes();
+        }
+    }
```

Comment:
Looks good.  What happens if the flatten function gets a flat file as the param?  Looks like something easy to check by sending the flat binary into the test, and see if the method just returns it.

Reply:
@ptrelease have attempted to unit test but when flattening an additional byte of information is added and modified by PDFBox for tracking/flagging, the resulting file of flattening an already flattened PDF is an unchanged PDF that is flat. However, there is an additional byte in the result PDF that is used by PDFBox. 

The PDFBox Library also has a unit tests of there own covering this check in pdacroformflattentest.java, where the PDF before and after flattening are rendered and content is compared.

Reply:
Exception test looks great.  And though a test wasn't viable, al has added a comment on ln65 that answers my question